### PR TITLE
refactor: [M3-6437] - MUI v5 Migration - Features > Billing

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
As an initial test run for [MUI Grid v2](https://mui.com/material-ui/migration/migration-grid-v2/), I refactored the Billing section with `Grid2` to gauge the difficulty. There are a few differences, especially with the custom styles we pass to the grid component, but overall it was a 1-to-1 replacement. I put together a temporary storybook page showing our original Grid vs Grid2.

> **Note**: See documentation for changes https://mui.com/material-ui/migration/migration-grid-v2/

## Major Changes 🔄
- Removed `makeStyles` and old JSS patterns for `styled/sx` API
- The cards on the billing landing page are all uniform now, with matching borders and spacing
- Added basic storybook page comparing both grids

**Files**
- `BillingDetail.tsx`: Simplified the amount of `Grid` components we were using. Created several re-useable components for uniformity across cards.
- `BillingActivityPanel.tsx`: Wrapped in a `Grid` due to the new way we've laid out our JSX in `BillingDetail`.
- `BillingSummary.tsx`: Moved `PromoDisplay` to its own file. Replaced classes with `sx` equivalent.
- `ContactInformation.tsx`: Refactored entire file using re-useable components and new API. Removed `Grid` where it was not needed, and replaced with `Box`.
- `UpdateContactInformationForm.tsx`: Removed the conditional re-rendering we were trying to do with `updateFor` and `RenderGuard` - There didn't seem to be perf difference, but this is something we can look at closer.
- `PaymentInformation.tsx`: Was able to really cleanup this file by re-using components from `BillingDetail`
- `InvoiceDetail.tsx:` Replaced several `Grid`'s with `Box` component. Cleaned up styles, spacing and removed extra padding around things.

## Places to Review 📷
| Landing      | Make a Payment | Add Payment Method | Edit Billing Contact |
| ----------- | ---------------- | ----------- | ----------- |
| Screen Shot 2023-03-28 at 4 15 54 PM | Screen Shot 2023-03-28 at 11 08 02 AM | Screen Shot 2023-03-24 at 10 56 15 AM | Screen Shot 2023-03-24 at 10 55 57 AM 

| Invoice Before      | Invoice After |
| ----------- | ----------- |
|Screen Shot 2023-03-28 at 4 02 48 PM|Screen Shot 2023-03-28 at 4 02 36 PM|

^ Overall spacing when inspecting is much nicer than before using `Box` and flexbox.

## How to test 🧪
**Routes:**
- http://localhost:3000/account/billing/
- http://localhost:3000/account/billing/edit
- http://localhost:3000/account/billing/add-payment-method
- http://localhost:3000/account/billing/invoices/{number}
- http://localhost:3000/account/billing/make-payment

### CODE REVIEW

**Review by bnussman-akamai — COMMENTED** [review]
> cards on the billing landing page are all uniform now

Thank you!! This has always bugged me

File: packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/UpdateContactInformationForm/UpdateContactInformationForm.tsx
Comment (bnussman-akamai) [inline packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/UpdateContactInformationForm/UpdateContactInformationForm.tsx:1]: Looks like we lost some spacing here

| Before | After |
|--------|-------|
| Screenshot 2023-03-29 at 10 43 14 AM      | Screenshot 2023-03-29 at 10 43 05 AM     |

File: packages/manager/src/features/Billing/BillingPanels/PaymentInfoPanel/AddPaymentMethodDrawer/AddPaymentMethodDrawer.tsx
Comment (dwiley-akamai) [inline packages/manager/src/features/Billing/BillingPanels/PaymentInfoPanel/AddPaymentMethodDrawer/AddPaymentMethodDrawer.tsx:1]: The Google Pay & PayPal icons got shifted right a bit and extend past the horizontal line:

<details>
<summary>Screenshot</summary>

Screenshot 2023-03-29 at 4 03 23 PM
</details>

**Review by dwiley-akamai — COMMENTED** [review]
Thanks for the Grid story in Storybook, that definitely helps compare the two Grid versions.

File: packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/UpdateContactInformationForm/UpdateContactInformationForm.tsx
Comment (jaalah-akamai) [reply packages/manager/src/features/Billing/BillingPanels/ContactInfoPanel/UpdateContactInformationForm/UpdateContactInformationForm.tsx:1]: Thanks, should be good now

File: packages/manager/src/features/Billing/BillingPanels/PaymentInfoPanel/AddPaymentMethodDrawer/AddPaymentMethodDrawer.tsx
Comment (jaalah-akamai) [reply packages/manager/src/features/Billing/BillingPanels/PaymentInfoPanel/AddPaymentMethodDrawer/AddPaymentMethodDrawer.tsx:1]: Should be fixed now

**Comment by jaalah-akamai** [comment]
@hana-linode [I addressed the spacing](https://github.com/linode/manager/pull/8939#pullrequestreview-1364129785) with the new Grid instead of margins. Here are some screenshots:
| SS | 16px top/bottom gap | 8px combined between cards |
|--|--|--|
| Screen Shot 2023-03-30 at 11 13 46 AM | Screen Shot 2023-03-30 at 11 13 59 AM | Screen Shot 2023-03-30 at 11 14 24 AM

**Review by bnussman-akamai — APPROVED** [review]
Looks good!! 🎉

That one conflicting file you have can be deleted completely. It is our old PayPal implementation.

**Review by mjac0bs — APPROVED** [review]
> The font size for the PayPal notice is also bigger. However, since PayPal has been available for recurring payments for a while now, I'm not opposed to removing that notice all together if the team agrees.

It looks like it's been [>1 year since that PayPal notice has been around](https://github.com/linode/manager/pull/8009). I'd support removing it altogether. Agreed with @hana-linode that the larger font is noticeable and we should adjust it if we don't remove the notice, though.

Other than that, I haven't been able to find any regressions or differences beyond what has been already pointed out and addressed. I used the MSW and modified the billing source to `akamai` in `account.ts` to check the minor UI differences for those users, too, and UI looked good there.

Update: Just noticed something minor -- "Make a Payment Immediately", which is displayed when the mocks are on, has some alignment issues and is no longer sentence case. Still going to approve, pending that is fixed.

This branch with MSW:
image

Prod:
image

**Comment by jaalah-akamai** [comment]
Fixed: Screen Shot 2023-04-03 at 9 05 30 PM

**Pull request merged**